source/RialtoGStreamerMSEAudioSink.cpp

```diff
@@ -66,6 +67,74 @@ static GstStateChangeReturn rialto_mse_audio_sink_change_state(GstElement *eleme
+    const gchar *strct_name = gst_structure_get_name(structure);
+    if (strct_name)
+    {
+        if (g_str_has_prefix(strct_name, "audio/mpeg") || g_str_has_prefix(strct_name, "audio/x-eac3") ||
+            g_str_has_prefix(strct_name, "audio/eac3") || g_str_has_prefix(strct_name, "audio/x-ac3") ||
+            g_str_has_prefix(strct_name, "audio/ac3"))
+        {
+            gint sample_rate = 0;
+            gint number_of_channels = 0;
+            gst_structure_get_int(structure, "rate", &sample_rate);
+            gst_structure_get_int(structure, "channels", &number_of_channels);
+            firebolt::rialto::AudioConfig audioConfig{static_cast<uint32_t>(number_of_channels),
+                                                      static_cast<uint32_t>(sample_rate),
+                                                      {}};
+            if (g_str_has_prefix(strct_name, "audio/mpeg"))
+            {
+                return firebolt::rialto::IMediaPipeline::MediaSource(-1, "audio/mpeg", audioConfig);
+            }
+            else
+            {
+                return firebolt::rialto::IMediaPipeline::MediaSource(-1, "audio/x-eac3", audioConfig);
+            }
+        }
+        else if (g_str_has_prefix(strct_name, "audio/x-opus"))
+        {
+            guint32 sample_rate = 48000;
+            guint8 number_of_channels, streams, stereo_streams, channel_mapping_family;
+            guint8 channel_mapping[256];
+            GstBuffer *id_header;
+            guint16 pre_skip = 0;
+            gint16 gain = 0;
+            if (gst_codec_utils_opus_parse_caps(caps, &sample_rate, &number_of_channels, &channel_mapping_family,
+                                                &streams, &stereo_streams, channel_mapping))
+            {
+                id_header = gst_codec_utils_opus_create_header(sample_rate, number_of_channels, channel_mapping_family,
+                                                               streams, stereo_streams, channel_mapping, pre_skip, gain);
+                std::vector<uint8_t> codec_specific_config;
+                GstMapInfo lsMap;
+                if (gst_buffer_map(id_header, &lsMap, GST_MAP_READ))
+                {
+                    codec_specific_config.assign(lsMap.data, lsMap.data + lsMap.size);
+                    gst_buffer_unmap(id_header, &lsMap);
+                }
+                else
+                {
+                    GST_ERROR_OBJECT(id_header, "Failed to read opus header details from a GstBuffer!");
+                }
+                gst_buffer_unref(id_header);
+
+                firebolt::rialto::AudioConfig audioConfig{number_of_channels, sample_rate, codec_specific_config};
+                return firebolt::rialto::IMediaPipeline::MediaSource(-1, "audio/x-opus", audioConfig);
+            }
+            else
+            {
+                GST_ERROR_OBJECT(caps, "Failed to parse opus caps!");
+            }
+        }
+
```

Comment:
For debugging would it be helpful to have an else clause on here?

else
{
    log_error("Unknown MIME type");
}